File: nodejs/debugger.js
# sleep
function sleep(milliSeconds) { 
    var startTime = new Date().getTime(); 
    while (new Date().getTime() < startTime + milliSeconds);
};
sleep(10000);

# output log
console.log("test");

# interrupt
debugger

# run shell
const exec = require('child_process').exec;
var yourscript = exec('env', (error, stdout, stderr) => {
            console.log(stdout);
            console.log(stderr);
});

const exec = require('child_process').exec;
var yourscript = exec('ls -al .', (error, stdout, stderr) => {
            console.log(stdout);
            console.log(stderr);
});

# get stack info

# profile

#read config from external

#format by template

# pid

# cwd

File: go/debugger.go
package main;

import "os";
import "log";
import "fmt";
import "time";
import "os/exec";
import "runtime/debug"

func main(){
  //sleep 1s
  time.Sleep(time.Duration(1)*time.Second)

  //ouput log
  print("hello")
  println(" davis")

  //interrupt

  //run shell
  out, err := exec.Command("env").Output()
	if err != nil {
	   log.Fatal(err)
	}
	fmt.Printf("The env is %s\n", out)

  out, err = exec.Command("ls","-al",".").Output()
	if err != nil {
	   log.Fatal(err)
	}
	fmt.Printf("The files is %s\n", out)

  //get stack info
  debug.PrintStack();

  //profile

  //read config from external

  //format by template

  //pid
  fmt.Printf("The pid is %d\n", os.Getppid())

  //goruntine

  //cpu usage

  //cwd
  dir, err := os.Getwd()
	if err != nil {
		log.Fatal(err)
	}
  fmt.Println(dir)
}
